iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1233,6 +1277,13 @@ pub struct DirectAddrInfo {
     ///
     /// [`Endpoint::add_node_addr`]: crate::endpoint::Endpoint::add_node_addr
     pub last_alive: Option<Duration>,
+    /// A [`HashMap`] of [`Source`]s to [`Duration`]s.
+    ///
+    /// Duration is calculated as the time passed from when this source was last recorded and
+    /// the instant this object was created.
```

Comment:
```suggestion
    /// The [`Duration`] indicates the elapsed time since this source last
    /// recorded this address.
```
Somehow I find that easier to parse. The other descriptions also do not explicitly say this is elapsed time from when this object is returned.